Dockerfile

```diff
@@ -0,0 +1,21 @@
+# Use vLLM's vllm-openai server image as the base
+FROM vllm/vllm-openai:latest
+
+# Set working directory
+WORKDIR /workspace
+
+# Install necessary build dependencies for sentencepiece
+RUN apt-get update && apt-get install -y \
+    pkg-config \
+    cmake \
+    build-essential
+
+# Copy functionary code and requirements into workspace
+COPY . .
+
+# Install additional Python dependencies
+RUN --mount=type=cache,target=/root/.cache/pip \
+    python3 -m pip install -r requirements.txt
+
+# Override the VLLM entrypoint with the functionary server
+ENTRYPOINT ["python3", "server_vllm.py", "--model", "meetkai/functionary-small-v3.2", "--host", "0.0.0.0", "--max-model-len", "8192"]
```

Comment:
Can we allow user-defined options at container runtime? Maybe something like `CMD []`